ARTstor.js

```diff
@@ -1,131 +1,817 @@
+
+            } else if (type == "4") {
+                //search notes
+                var aType = searchTerm.aType;
+                serviceURL = getServiceUrlRoot(url) + pageType + "/" + params[2] + "/" + startIdx + "/" + pageSize + "/" + sortOrder + "?type=" + type + "&kw=" +
+                    kw + "&origKW=" + origKW + "&aType=" + aType + "&order=" + order + "&tn=1";
+
+            } else {
+                //KW search type=6, PC, All Coll, Inst Coll, Adv Srch
+                var collectionTit = decrypt(params[3]);
+                var collectionTitle = collectionTit.split(":");
+                var name = escape(collectionTitle[0]);
+                serviceURL = getServiceUrlRoot(url) + pageType + "/" + params[2] + "/" + startIdx + "/" + pageSize + "/" + sortOrder + "?type=" + type + "&kw=" +
+                    kw + "&origKW=" + origKW + "&geoIds=" + searchTerm.geoIds + "&clsIds=" + searchTerm.clsIds + "&collTypes=" + searchTerm.collTypes + "&id=" +
+                    searchTerm.id + "&name=" + name + "&bDate=" + searchTerm.bDate +
+                    "&eDate=" + searchTerm.eDate + "&dExact=" + searchTerm.dExact + "&order=" + order + "&isHistory=false&prGeoId=" + searchTerm.prGeoId + "&tn=1";
+            }
+            //p=escape(p);
+            break;
+        case "collaboratoryfiltering":
+            var serviceURL = getServiceUrlRoot(url) + pageType + "/" + params[2] + "/thumbnails/" + startIdx + "/" + pageSize + "/" + sortOrder + "?collectionId=" + params[8];
+            break;
+        case "imagegroup": // image group
+        case "collections":
+        case "categories":
+        case "cluster":
+        default:
+            serviceURL = getServiceUrlRoot(url) + pageType + "/" + contentId + "/thumbnails/" + startIdx + "/" + pageSize + "/" + sortOrder;
+    }
+    return serviceURL;
+}
+
+function getServiceUrlRoot(url) {
+    var serviceRoot = getServerUrl(url) + "/secure/";
+    return serviceRoot;
+
+}
+
+function getFileRoot(url) {
+    var fileRoot = getServerUrl(url).substring(0, getServerUrl(url).lastIndexOf('/'));
+    return fileRoot;
+}
+
+/**
+    decodeSearchData is an helper function to process the search result page.
+    It converts the search url parameter into arrary of parameter values.
+
+    It converts
+    type=6&kw=stimson hall|all#and,1925|all&geoIds=&clsIds=&collTypes=&id=all&bDate=-2019&eDate=2030&dExact=1&prGeoId=&origKW=stimson hall|all#and,1925|all
+    to  {
+             "type": 6
+             "kw": "stimson hall|all#and,1925|all"
+             "geoIds": ""
+             "clsIds": ""
+             "collTypes": ""
+             "id": "all"
+             "bDate": "-2019"
+             "eDate": "2030"
+             "dExact": "1"
+             "prGeoId": ""
+             "origKW": "stimson hall|all#and,1925|all"
+         }
+**/
+function decodeSearchData(str) {
+    var searchParam = str.replace(/(&gt;)/g, ":");
```

Comment:
are you intentionally replacing this with incorrect decoding? Is that the only HTML entity that can be present? Maybe it's safer to use `ZU.unescapeHTML`, which unescapes all HTML entities.